code/modules/admin/verbs/individual_logging.dm

```diff
@@ -46,22 +46,27 @@
 	var/log_source = M.logging;
 	if(source == LOGSRC_CLIENT && M.client) //if client doesn't exist just fall back to the mob log
 		log_source = M.client.player_details.logging //should exist, if it doesn't that's a bug, don't check for it not existing
-
+	var/list/concatenated_logs = list()
 	for(var/log_type in log_source)
 		var/nlog_type = text2num(log_type)
 		if(nlog_type & ntype)
-			var/list/reversed = log_source[log_type]
-			if(islist(reversed))
-				reversed = reverseRange(reversed.Copy())
-				for(var/entry in reversed)
-					dat += "<font size=2px><b>[entry]</b><br>[reversed[entry]]</font><br>"
-			dat += "<hr>"
+			var/list/all_the_entrys = log_source[log_type]
+			for(var/entry in all_the_entrys)
+				concatenated_logs += "<b>[entry]</b><br>[all_the_entrys[entry]]"
+	if(length(concatenated_logs))
+		sortTim(concatenated_logs, cmp = /proc/cmp_text_dsc) //Sort by timestamp.
+		dat += "<font size=2px>"
+		dat += concatenated_logs.Join("<br>")
+		dat += "</font>"
 
-	usr << browse(dat, "window=invidual_logging_[key_name(M)];size=600x480")
+	var/datum/browser/popup = new(usr, "invidual_logging_[key_name(M)]", "Individual Logs", 600, 600)
+	popup.set_content(dat.Join())
+	popup.open()
 
 /proc/individual_logging_panel_link(mob/M, log_type, log_src, label, selected_src, selected_type)
 	var/slabel = label
 	if(selected_type == log_type && selected_src == log_src)
 		slabel = "<b>\[[label]\]</b>"
-
+	//This is necessary because num2text drops digits and rounds on big numbers. If more defines get added in the future it could break again.
+	log_type = num2text(log_type, 10)
```

Comment:
Magic number, please move out to a define.

Reply:
How would I do that?

Reply:
At the top -- `#define WHATEVER_THE_NUMBER_IS_FOR 10`, then at the end--`#undef WHATEVER_THE_NUMBER_IS_FOR`.

Reply:
This is not a magic number, this is a parameter for significant digits in a byond proc:
<http://www.byond.com/docs/ref/index.html#/proc/num2text>

Reply:
```suggestion
	log_type = num2text(log_type, MAX_BITFLAG_DIGITS)
```
and then in the flag defines file:
```
/// 33554431 (2^24 - 1) is the maximum value our bitflags can reach.
#define MAX_BITFLAG_DIGITS 8
```

Reply:
Done.